packages/kit/src/hooks/useFlowRevertibleRandom.ts

```diff
@@ -0,0 +1,53 @@
+import {UseQueryOptions, UseQueryResult} from "@tanstack/react-query"
+import {useFlowQuery} from "./useFlowQuery"
+
+export interface RevertibleRandomResult {
+  blockHeight: string
+  value: string
+}
+
+export interface UseFlowRevertibleRandomArgs {
+  min?: string
+  max: string
+  query?: Omit<UseQueryOptions<unknown, Error>, "queryKey" | "queryFn">
+}
+
+export function useFlowRevertibleRandom({
+  min = "0",
+  max,
+  query: queryOptions = {},
+}: UseFlowRevertibleRandomArgs): UseQueryResult<RevertibleRandomResult, Error> {
+  const mergedQuery = {
+    staleTime: Infinity,
+    refetchOnWindowFocus: false,
+    ...queryOptions,
+  }
+
+  const result = useFlowQuery({
+    cadence: `
+      access(all) struct RevertibleRandomResult {
+        access(all) let blockHeight: UInt64
+        access(all) let value: UInt256
+
+        init(blockHeight: UInt64, value: UInt256) {
+          self.blockHeight = blockHeight
+          self.value = value
+        }
+      }
+
+      access(all) fun main(min: UInt256, max: UInt256): RevertibleRandomResult {
+        pre {
+          min < max: "Invalid random range - max must be greater than min"
+        }
+        return RevertibleRandomResult(
+          blockHeight: getCurrentBlock().height,
+          value: revertibleRandom<UInt256>(modulo: max - min + 1)
+        )
+      }
```

Comment:
Here's what that script would look like where the script generates a given count of random results

```suggestion
access(all) struct RevertibleRandomResult {
    access(all) let blockHeight: UInt64
    access(all) let value: UInt256

    init(blockHeight: UInt64, value: UInt256) {
        self.blockHeight = blockHeight
        self.value = value
    }
}

access(all) fun main(min: UInt256, max: UInt256, count: Int): [RevertibleRandomResult] {
    pre {
        min < max: "Invalid random range - max must be greater than min"
        0 < count: "Invalid count - must request at least one random value at a time"
    }
    let currentHeight = getCurrentBlock().height
    let results: [RevertibleRandomResult] = []
    for i in InclusiveRange(1, count) {
        results.append(RevertibleRandomResult(
            blockHeight: currentHeight,
            value: revertibleRandom<UInt256>(modulo: max - min + 1)
        ))
    }
    return results
}
```